web/src/app/browser/src/keymanEngine.ts

```diff
@@ -241,6 +242,57 @@ export default class KeymanEngine extends KeymanEngineBase<BrowserConfiguration,
     }
   }
 
+  /**
+   * Get an associative array of keyboard identification strings
+   *   This was defined as an array, so is kept that way, but
+   *   Javascript treats it as an object anyway
+   *
+   * @param       {Object}    Lstub      Keyboard stub object
+   * @param       {Object}    Lkbd       Keyboard script object
+   * @return      {Object}               Copy of keyboard identification strings
+   *
+   */
+  private _GetKeyboardDetail = function(Lstub: KeyboardStub, Lkbd: any /* KeyboardScriptObject */) { // I2078 - Full keyboard detail
+    var Lr={};
+    Lr['Name'] = Lstub['KN'];
+    Lr['InternalName'] =  Lstub['KI'];
+    Lr['LanguageName'] = Lstub['KL'];  // I1300 - Add support for language names
+    Lr['LanguageCode'] = Lstub['KLC']; // I1702 - Add support for language codes, region names, region codes, country names and country codes
+    Lr['RegionName'] = Lstub['KR'];
+    Lr['RegionCode'] = Lstub['KRC'];
+    Lr['CountryName'] = Lstub['KC'];
+    Lr['CountryCode'] = Lstub['KCC'];
+    Lr['KeyboardID'] = Lstub['KD'];
+    Lr['Font'] = Lstub['KFont'];
+    Lr['OskFont'] = Lstub['KOskFont'];
+    Lr['HasLoaded'] = !!Lkbd;
+
+    Lr['IsRTL'] = Lkbd ? !!Lkbd['KRTL'] : null;
+    return Lr;
+  }
+
+  /**
+   * Get API-friendly array of available keyboard stubs
+   *
+   * @return   {Array}     Array of available keyboards
+   *
+   */
+  getKeyboards() {
```

Comment:
```suggestion
  public getKeyboards() {
```

Helping to indicate that this is a public API

Reply:
I've added a tracking item on #8560 regarding this.  While I've made the change suggested here, I get the feeling that this and the other comment are intended to be generalized beyond the scope of this PR.

Reply:
> I've added a tracking item on #8560 regarding this. While I've made the change suggested here, I get the feeling that this and the other comment are intended to be generalized beyond the scope of this PR.

Yes, that's a good feeling to have here :grin: